src/context.jl

```diff
@@ -1,24 +1,60 @@
 # OpenCL.Context
 
+const _ctx_id_seen = Dict{CL_context, Bool}()
```

Comment:
What exactly do you mean by a device reset?
But yeah, I think what we actually need is counting also the julia references.. Will update the PR!